### PR TITLE
Access Tokens Per Device - JWT

### PR DESCRIPTION
**Pull request by ruthryi**
Fixes: 
- https://github.com/wso2-enterprise/wso2-apim-internal/issues/4033

### CODE REVIEW

File: en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # Access Tokens Per Device
 
-WSO2 API Manager returns the same token repeatedly if a valid token exists for the requesting Application, on behalf of the user. However, the latter mentioned scenario becomes an issue if the same user is using the same Application in two devices (e.g., If you have two instances of the same Application running on your iPhone and iPad, and your iPhone already has a token on behalf of you, your iPad will get the same token if you requested for it within the same validity period. Therefore, if one of your devices revoke this token (e.g., revoke on logout), the token that you obtained for your other device becomes invalid as the devices use the identical tokens.
+In WSO2 API Manager, the behavior for generating JWT tokens with the same application keys and scopes has changed. Unlike the previous approach with Opaque tokens, WSO2 API Manager no longer returns the same existing valid JWT token for each token request. Additionally, if a valid token already exists for the requesting application and user, generating a new token will result in the revocation of the previous token associated with that application and user.
```
[the remark this answers is not in the document]
Comment (ruthryi) [reply en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md]: Done. Please check.

File: en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
+
+To address this, WSO2 API Manager employs a mechanism that leverages [OAuth2.0 Scopes]({{base_path}}/design/api-security/oauth2/oauth2-scopes/fine-grained-access-control-with-oauth-scopes/#fine-grained-access-control-with-oauth-scopes) to generate unique Access Tokens for each device using the same application. This mechanism allows users to request tokens for different scopes. To achieve this, you are required to prefix the scope names with the string `device_`. Notably, WSO2 API Manager applies special treatment to scopes that bear this particular prefix, by bypassing the usual validations typically performed when issuing tokens associated with scopes.
```
Comment (dushaniw) [inline en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md:5]: Before mentioning this, we need to add the following part.

The latter mentioned scenario becomes an issue if the same user is using the same Application in two devices (e.g., If you have two instances of the same Application running on your iPhone and iPad, and your iPhone already has a token on behalf of you, your iPad will get a new token if you requested for it within the same validity period and with the same scopes. Therefore, the token that you obtained for your other device iPhone becomes revoked.

**Review by dushaniw — CHANGES_REQUESTED** [review]

File: en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md (hunk 2 above)
Comment (ruthryi) [reply en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md:7]: Added a modified version of this as a paragraph. Please check.

File: en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
+In WSO2 API Manager, the behavior for generating JWT tokens with the same application keys and scopes is different to the previous approach with Opaque tokens. For JWT tokens, WSO2 API Manager no longer returns the same existing valid JWT token for each token request if the associated application, user and requested scopes are the same. Additionally, if a valid token already exists for the requesting application, user and requested scopes, generating a new token will result in the revocation of the previous token associated with that application and user.
+
+However, this change in behavior is particularly relevant when considering scenarios where the same user is utilizing the same application across multiple devices. For example, if a user has the same application installed on both their iPhone and iPad, and a token has been generated on the iPhone, requesting a token on the iPad within the same validity period could result in obtaining the same token. Consequently, if one of these devices revokes the token, such as during a logout operation, the token obtained for the other device will also become invalid. This is because both devices are utilizing identical tokens.
```
Comment (dushaniw) [inline en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md:5]: @ruthryi this is wrong. Both wont be using identical tokens.

The idea  is, if token is generated by iPhone and if iPad is generating the token for same application, user and scopes, then iPhone's token will be revoked. iPad will also get a new token.

Comment (ruthryi) [reply en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md]: Done

File: en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
+In WSO2 API Manager, the behavior for generating JWT tokens with the same application keys and scopes is different to the previous approach with Opaque tokens. For JWT tokens, WSO2 API Manager no longer returns the same existing valid JWT token for each token request if the associated application, user and requested scopes are the same. Additionally, if a valid token already exists for the requesting application, user and requested scopes, generating a new token will result in the revocation of the previous token associated with that application and user.
+
+However, this change in behavior addresses scenarios where a user employs the same application across multiple devices. For example, if a token is generated on an iPhone and the same application, user, and scopes prompt the iPad to request a token within the same validity period, the iPhone's token will be revoked, while the iPad will obtain a new one. This process prevents overlap of tokens and ensures distinct tokens per device, even for the same application and user credentials.
```
Comment (dushaniw) [inline en/docs/consume/manage-application/generate-keys/obtain-access-token/access-tokens-per-device.md:5]:
```suggestion
However, this change in behavior addresses scenarios where a user employs the same application across multiple devices. For example, if a token is generated on an iPhone and the same application, user, and scopes prompt the iPad to request a token within the same validity period, the iPhone's token will be revoked, while the iPad will obtain a new one. This process prevents overlap of tokens and does not ensure distinct tokens per device, even for the same application and user credentials.
```

**Pull request merged**